libyul/Utilities.cpp

```diff
@@ -32,11 +33,41 @@ using namespace yul;
 u256 yul::valueOfNumberLiteral(Literal const& _literal)
 {
 	assertThrow(_literal.kind == LiteralKind::Number, OptimizerException, "");
+
 	std::string const& literalString = _literal.value.str();
 	assertThrow(isValidDecimal(literalString) || isValidHex(literalString), OptimizerException, "");
 	return u256(literalString);
 }
 
+u256 yul::valueOfStringLiteral(Literal const& _literal)
+{
+	assertThrow(_literal.kind == LiteralKind::String, OptimizerException, "");
```

Comment:
Probably these assertions should be ``YulAssertion``s instead of ``OptimizerException``s anyways.